src/kraken/core/system/context.py

```
@@ -279,7 +291,7 @@ def get_build_graph(self, targets: Sequence[str | Task] | None) -> TaskGraph:
         """
 
         if targets is None:
-            tasks = self.resolve_tasks(None)
+            tasks = self.resolve_tasks([":"])
```

> either all default tasks

I think that means we must default to `[":", ":**:"]` as you are using it in `_load_build_state()`.

Basically it boils down to making `resolve_tasks()` accept `None` again and making it behave like `[":", ":**:"]` :)

---

Well, I'm hiding dust under the carpet here.
`:**:` raises an exception in case there is no sub-project. That's why I removed the "None becomes `[":", ":**:"]`" conversion from `resolve_tasks()`, and moved it into `main.py`, which is able to ignore the exception when `:**:` is requested (but not in case a different address is requested).

But I think you are right, this logic should be taken into account here as well.
I'll see what sensible solution I can do